AdyenCard/Components/GiftCardComponent/PartialPaymentMethodDetails.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright (c) 2023 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+#if canImport(AdyenEncryption)
+    import AdyenEncryption
+#endif
+import Foundation
+
+/// Describes the details a partial payment method (e.g. a gift card) needs to provide.
+public protocol PartialPaymentMethodDetails: PaymentMethodDetails {
+    
+    /// The payment method type.
+    var type: PaymentMethodType { get }
+    
+    /// The encrypted card number.
+    var encryptedCardNumber: String { get }
+    
+    /// The encrypted security code.
+    var encryptedSecurityCode: String { get }
+}
+
+/// Contains the details provided by the gift card component with meal voucher payment method.
+public struct MealVoucherDetails: PartialPaymentMethodDetails {
```

Comment:
Maybe better to move to its own file for better visibility and consistency with `GiftCardDetails`